app/src/main/java/com/pwr266521/bmiapp/MainActivity.kt

```diff
@@ -1,11 +1,193 @@
 package com.pwr266521.bmiapp
 
-import androidx.appcompat.app.AppCompatActivity
+import android.content.Intent
+import android.graphics.Color
+import android.os.Build
 import android.os.Bundle
+import android.view.MenuItem
+import android.view.View
+import android.widget.Button
+import android.widget.EditText
+import android.widget.PopupMenu
+import android.widget.TextView
+import androidx.annotation.RequiresApi
+import com.pwr266521.bmiapp.domain.BMIHistory
+import com.pwr266521.bmiapp.util.BMIUtil.BMI_KEY
+import com.pwr266521.bmiapp.util.BMIUtil.bmiHistory
+import com.pwr266521.bmiapp.util.BMIUtil.calculateBMI
+import java.time.LocalDateTime
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.Toolbar
 
 class MainActivity : AppCompatActivity() {
+    companion object {
+        private const val BMIDelimiter = ":"
+        private var isImperialUnit = false
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        val editTextWeight: EditText = findViewById(R.id.editTextWeight)
+        val weightUnitVIew: TextView = findViewById(R.id.kg_view)
+        val heightUnitView: TextView = findViewById(R.id.cm_view)
+        val inchesTextHeight: TextView = findViewById(R.id.inchesTextHeight)
+        val inchesTextView: TextView = findViewById(R.id.inchesTextView)
+        val editTextHeight: EditText = findViewById(R.id.editTextHeight)
+        val buttonCalculate: Button = findViewById(R.id.buttonCalculate)
+        val textViewResult: TextView = findViewById(R.id.textViewResult)
+        val buttonDetails: Button = findViewById(R.id.detailsButton)
+        val toolbar: Toolbar = findViewById(R.id.toolbar)
+
+        buttonCalculate.setOnClickListener {
+            processBMICalculation(
+                editTextWeight,
+                editTextHeight,
+                textViewResult,
+                editTextHeight,
+                buttonDetails
+            )
+        }
+        buttonDetails.setOnClickListener { processBMIDetails(textViewResult) }
+
+
+        customizeToolbar(toolbar)
+        toolbar.setOnClickListener {
+            showDropdownMenu(
+                it,
+                weightUnitVIew,
+                heightUnitView,
+                inchesTextView,
+                inchesTextHeight
+            )
+        }
+    }
+
+    private fun customizeToolbar(toolbar: Toolbar) {
+        setSupportActionBar(toolbar)
+        toolbar.setTitleTextColor(Color.WHITE)
+        supportActionBar?.title = "⋮"
```

Comment:
Lepszą opcją byłoby zrobienie przycisku na top barze, zamiast dodawania tytułu wyglądającego jak przycisk i dodawania listenera na cały top bar. W tym obrazku który mi wysłałeś ten przycisk jest po prawej stronie top bara i wg oczekiwane jest to żeby był to przycisk.